### PR TITLE
Switch JSHint to ESLint style checker.

### LINKED ISSUE #62: Linting: line length
Currently there are some lines >80 columns. If this is supposed to be the limit, it should be checked in the CI.

cc: @jmuk @tbetbetbe

### PR DESCRIPTION
**Pull request by jmuk**
This also includes style fixes found by ESLint.
Fixes #62.

### CODE REVIEW

File: lib/api_repo.js [hunk 1]
```diff
@@ -18,14 +18,14 @@
...
-var child_process = require('child_process');
+var execFile = require('child_process').execFile;
 var config = require('./config');
 var fs = require('fs-extra');
 var glob = require('glob');
 var packager = require('./packager');
 var path = require('path');
 var pbjs = require('protobufjs/cli/pbjs');
-var pbjs_util = require('protobufjs/cli/pbjs/util');
+var pbjs_util = require('protobufjs/cli/pbjs/util'); // eslint-disable-line
```
Comment (geigerj) [inline lib/api_repo.js]: Why does eslint complain about this line?

File: lib/api_repo.js [hunk 2]
```diff
@@ -429,11 +451,11 @@ function collectProtoDeps(includePath, protoFile, done) {
  * This is a sub task of 'ProtoBased' nodeJS module in _buildProtos, but
  * extracted as a method for the testing.
  *
- * @param {string} outDir - the top of the output directory.
- * @param {[][]string} fileList - the list of the lists of the .proto files
- *   to be copied. Each item is the list of .proto files calculated by
+ * @param {Array.<string[]>} fileLists - the list of the lists of the .proto
```
Comment (geigerj) [inline lib/api_repo.js]: Why mixed notation for array? I would expect either `Array.<Array.<string>>` or `string[][]`

File: lib/api_repo.js (hunk 1 above)
Comment (geigerj) [inline lib/api_repo.js]: If it's just the snake_case, should we change it to camelCase?

File: lib/packager.js [hunk 3]
```diff
@@ -590,6 +605,7 @@ function checkedCopy(src, dst)  {
  * @param {string} dst the path of the expanded output
  * @param {Object} params object containing the named parameter values
  * @param {function(Error, string)} done is called with the rendered template
+ * @return {function} a funciton which does the expanding task.
```
Comment (geigerj) [inline lib/packager.js]: sp(funciton)

**Comment by geigerj** [comment]
just nits, LGTM

File: lib/api_repo.js (hunk 1 above)
Comment (jmuk) [inline lib/api_repo.js]: It's for snake_case. Fixed by pbjsUtil.

File: lib/api_repo.js (hunk 2 above)
Comment (jmuk) [inline lib/api_repo.js]: Fixed (somehow ESLint fails to parse string[][], so Array.<Array.<string>>)

File: lib/packager.js (hunk 3 above)
Comment (jmuk) [inline lib/packager.js]: done

**Comment by jmuk** [comment]
Let me know when I can merge this -- I just don't want to make conflicts with ongoing PRs.

**Comment by geigerj** [comment]
You can go ahead and merge -- I am doing other self-service work at the moment and probably won't pick up the ongoing PR until tomorrow. I will rebase on this change when I pick it back up.

**Comment by jmuk** [comment]
Okay, merging.

**Pull request merged**